cni/network/stateless/main.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/Azure/azure-container-networking/cni/network"
+	"github.com/Azure/azure-container-networking/common"
+	"github.com/Azure/azure-container-networking/nns"
+	"github.com/Azure/azure-container-networking/platform"
+	"github.com/Azure/azure-container-networking/telemetry"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	hostNetAgentURL                 = "http://168.63.129.16/machine/plugins?comp=netagent&type=cnireport"
+	ipamQueryURL                    = "http://168.63.129.16/machine/plugins?comp=nmagent&type=getinterfaceinfov1"
+	pluginName                      = "CNI"
+	telemetryNumRetries             = 5
+	telemetryWaitTimeInMilliseconds = 200
+	name                            = "azure-vnet"
+	maxLogFileSizeInMb              = 5
+	maxLogFileCount                 = 8
+	stateless                       = true
+)
+
+// Version is populated by make during build.
+var version string
+
+// Command line arguments for CNI plugin.
+var args = common.ArgumentList{
+	{
+		Name:         common.OptVersion,
+		Shorthand:    common.OptVersionAlias,
+		Description:  "Print version information",
+		Type:         "bool",
+		DefaultValue: false,
+	},
+}
+
+// Prints version information.
+func printVersion() {
+	fmt.Printf("Azure CNI Version %v\n", version)
+}
+
+func rootExecute() error {
+	var (
+		config common.PluginConfig
+		tb     *telemetry.TelemetryBuffer
+	)
+
+	config.Version = version
+	config.Stateless = stateless
+
+	reportManager := &telemetry.ReportManager{
+		HostNetAgentURL: hostNetAgentURL,
+		ContentType:     telemetry.ContentType,
+		Report: &telemetry.CNIReport{
+			Context:          "AzureCNI",
+			SystemDetails:    telemetry.SystemInfo{},
+			InterfaceDetails: telemetry.InterfaceInfo{},
+			BridgeDetails:    telemetry.BridgeInfo{},
+			Version:          version,
+		},
+	}
+
+	cniReport := reportManager.Report.(*telemetry.CNIReport)
+
+	netPlugin, err := network.NewPlugin(
+		name,
+		&config,
+		&nns.GrpcClient{},
+		&network.Multitenancy{},
+	)
+	if err != nil {
+		network.PrintCNIError(fmt.Sprintf("Failed to create network plugin, err:%v.\n", err))
+		return errors.Wrap(err, "Create plugin error")
+	}
+
+	// Check CNI_COMMAND value
+	cniCmd := os.Getenv(cni.Cmd)
+
+	if cniCmd != cni.CmdVersion {
+		log.Logger.Info("Environment variable set", zap.String("CNI_COMMAND", cniCmd))
+
+		cniReport.GetReport(pluginName, version, ipamQueryURL)
+
+		var upTime time.Time
+		upTime, err = platform.GetLastRebootTime()
+		if err == nil {
+			cniReport.VMUptime = upTime.Format("2006-01-02 15:04:05")
+		}
+
+		defer func() {
+			if recover() != nil {
+				os.Exit(1)
+			}
+		}()
+
+		// Start telemetry process if not already started. This should be done inside lock, otherwise multiple process
+		// end up creating/killing telemetry process results in undesired state.
+		tb = telemetry.NewTelemetryBuffer()
+		tb.ConnectToTelemetryService(telemetryNumRetries, telemetryWaitTimeInMilliseconds)
```

Comment:
stateless cni should try to connect to telemetry socket and if it couldn't should skip and move forward.. should not start any telemetry service.. have to update/add existing or new api